libfuzzer.cc

```diff
@@ -68,8 +67,28 @@ std::vector<std::string>& args_global = *new std::vector<std::string>();
 
 bool setup_called = false;
 
+unsigned long long num_counters = 0;
+unsigned char* counters = NULL;
+
 }  // namespace
 
+NO_SANITIZE
+void _loc(unsigned long long idx) {
+  if (counters && idx < num_counters) {
+    counters[idx]++;
+  }
+}
+
+NO_SANITIZE
+void _reg(unsigned long long num) {
```

Comment:
Can we have better names for these functions? `_cmp` is kinda obvious to me, but `_loc` and `_reg` aren't.